app/models/user.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_many :posts
+  before_destroy :cleanup
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+
+  validates :first_name, presence: true, length: { maximum: 50 }
+  validates :last_name, length: { maximum: 50 }
+  validates :email, presence: true, length: { maximum: 255 },
```

Comment:
Indeed Devise has a basic validation but I was able to create a user with an invalid email, e.g. user@email. With that in mind, I will keep most of our validations, but I removed the password validation because it was redundant.